### DISCUSSION TITLE
Customisable trailing slash behaviour

### DISCUSSION THREAD

**Discussion by roduyemi** [TanStack/router#1469] [Ideas]
Hey @tannerlinsley!

We'd love to be able to add trailing slashes to urls.

We have a huge site (100's of millions of pages) which all have a trailing slash on them, as this was previously enforced. We'd like to move to TanStack router, however we don't want to go through the whole SEO process of needing to redirect all the old pages mopping up all the issues that causes along the way.

Would it be possible to make this a controllable option?

Thanks!

**Comment by schiller-manuel** [comment]
I added the following option in [v1.27.0](https://github.com/TanStack/router/releases/tag/v1.27.0):

https://tanstack.com/router/v1/docs/framework/react/api/router/RouterOptionsType#trailingslash-property

Is this what you need?

**Reply by roduyemi** [reply]
Yes - perfect! Will upgrade now. Thank you so much @schiller-manuel ✨
